client/my-sites/checkout/composite-checkout/payment-method-helpers.js

```diff
@@ -210,11 +215,142 @@ function WordPressLogo() {
+			method: 'POST',
+			redirect: 'manual',
+			credentials: 'include',
+			headers: { Accept: 'application/json', 'Content-Type': 'application/x-www-form-urlencoded' },
+			body: stringifyBody( bodyObj ),
+		} );
+	} catch ( error ) {
+		throw new Error( error );
+	}
+}
+
+async function createAccount( select ) {
+	const newSiteParams = JSON.parse( window.localStorage.getItem( 'siteParams' ) || '{}' );
+
+	const { email } = select( 'wpcom' )?.getContactInfo() ?? {};
+	const emailValue = email.value;
+	const recaptchaClientId = select( 'wpcom' )?.getRecaptchaClientId();
+	const isRecaptchaLoaded = typeof recaptchaClientId === 'number';
+
+	let recaptchaToken = undefined;
+	let recaptchaError = undefined;
+
+	if ( isRecaptchaLoaded ) {
+		recaptchaToken = await recordGoogleRecaptchaAction(
+			recaptchaClientId,
+			'calypso/signup/formSubmit'
+		);
+
+		if ( ! recaptchaToken ) {
+			recaptchaError = 'recaptcha_failed';
+		}
+	} else {
+		recaptchaError = 'recaptcha_didnt_load';
+	}
+
+	try {
+		const response = await wp.undocumented().createUserAndSite(
+			{
+				email: emailValue,
+				'g-recaptcha-error': recaptchaError,
+				'g-recaptcha-response': recaptchaToken || undefined,
+				is_passwordless: true,
+				signup_flow_name: 'onboarding-new',
+				validate: false,
+				ab_test_variations: getSavedVariations(),
+				new_site_params: newSiteParams,
+			},
+			null
+		);
+
+		response.bearer_token && createAccountCallback( response );
+		return response;
+	} catch ( error ) {
+		const errorMessage = error?.message ? getErrorMessage( error ) : error;
+		throw new Error( errorMessage );
+	}
+}
+
+function getErrorMessage( { error, message } ) {
+	switch ( error ) {
+		case 'already_taken':
+		case 'already_active':
+		case 'email_exists':
+			return i18n.translate( 'An account with this email address already exists.' );
+		default:
+			return message;
+	}
+}
+
+export async function wpcomTransaction( payload, isLoggedOutCart ) {
+	if ( isLoggedOutCart ) {
+		const { select, dispatch } = defaultRegistry;
+
+		return createAccount( select ).then( ( response ) => {
+			const siteIdFromResponse = get( response, 'blog_details.blogid', undefined );
+			const siteSlugFromResponse = get( response, 'blog_details.site_slug', undefined );
+
+			siteIdFromResponse && dispatch( 'wpcom' ).setSiteId( siteIdFromResponse );
+			siteSlugFromResponse && dispatch( 'wpcom' ).setSiteSlug( siteSlugFromResponse );
+
+			// If the account is already created(as happens when we are reprocessing after a transaction error), then
+			// the create account response will not have a site ID, so we fetch from state.
+			const siteId = siteIdFromResponse || select( 'wpcom' )?.getSiteId();
+			const newPayLoad = {
+				...payload,
+				cart: {
+					...payload.cart,
+					blog_id: siteId || '0',
+					cart_key: siteId || 'no-site',
+					create_new_blog: siteId ? false : true,
+				},
+			};
+
+			return wp.undocumented().transactions( newPayLoad || payload );
```

Comment:
Is there a case where `newPayLoad` will be falsy here? It's defined immediately above so I think it will always be set and we don't need the `||` fallback.

Reply:
Done.